cmd/kogito/command/shared/install_services.go

```diff
@@ -79,36 +153,81 @@ func (s servicesInstallation) SilentlyInstallOperatorIfNotExists() ServicesInsta
 	}
 	return s
 }
 
-func (s servicesInstallation) InstallKeycloak() ServicesInstallation {
+func (s *servicesInstallation) InstallKeycloak() ServicesInstallation {
 	if s.err == nil {
 		s.err = installKeycloak(s.client, s.namespace)
 	}
 	return s
 }
 
-func (s servicesInstallation) InstallKafka() ServicesInstallation {
+func (s *servicesInstallation) InstallKafka() ServicesInstallation {
 	if s.err == nil {
 		s.err = installKafka(s.client, s.namespace)
 	}
 	return s
 }
 
-func (s servicesInstallation) GetError() error {
+func (s *servicesInstallation) GetError() error {
 	return s.err
 }
 
-// ServicesInstallationBuilder creates the basic structure for services installation definition.
-func ServicesInstallationBuilder(client *kogitocli.Client, namespace string) ServicesInstallation {
-	return servicesInstallation{
-		namespace:         namespace,
-		client:            client,
-		operatorInstalled: false,
+func (s *servicesInstallation) installKogitoService(service v1alpha1.KogitoService, getService func() v1alpha1.KogitoService, messages serviceInfoMessages) error {
+	if s.err == nil {
+		log := context.GetDefaultLogger()
+		if !s.operatorInstalled { // depends on operator
+			log.Info(messages.notInstalledNoKogitoOperator)
+			return nil
+		}
+		if service == nil || reflect.ValueOf(service).IsNil() {
+			service = getService()
+		}
+		if err := kubernetes.ResourceC(s.client).Create(service); err != nil {
+			return fmt.Errorf(messages.errCreating, err)
+		}
+		log.Infof(messages.installed, s.namespace)
+		log.Infof(messages.checkStatus, service.GetName(), s.namespace)
+	}
+	return nil
+}
+
+// getDefaultDataIndex gets the default Data Index instance
+func (s *servicesInstallation) getDefaultDataIndex() v1alpha1.KogitoService {
+	return &v1alpha1.KogitoDataIndex{
+		ObjectMeta: metav1.ObjectMeta{Name: infrastructure.DefaultDataIndexName, Namespace: s.namespace},
+		Spec: v1alpha1.KogitoDataIndexSpec{
+			KogitoServiceSpec: defaultServiceSpec,
+			InfinispanMeta:    v1alpha1.InfinispanMeta{InfinispanProperties: v1alpha1.InfinispanConnectionProperties{UseKogitoInfra: true}},
+			KafkaMeta:         v1alpha1.KafkaMeta{KafkaProperties: v1alpha1.KafkaConnectionProperties{UseKogitoInfra: true}},
+		},
+		Status: v1alpha1.KogitoDataIndexStatus{KogitoServiceStatus: defaultServiceStatus},
+	}
+}
+
+// getDefaultJobsService gets the default Jobs Service instance
+func (s *servicesInstallation) getDefaultJobsService() v1alpha1.KogitoService {
+	return &v1alpha1.KogitoJobsService{
+		ObjectMeta: metav1.ObjectMeta{Name: infrastructure.DefaultJobsServiceName, Namespace: s.namespace},
+		Spec: v1alpha1.KogitoJobsServiceSpec{
+			KogitoServiceSpec: defaultServiceSpec,
+			InfinispanMeta:    v1alpha1.InfinispanMeta{InfinispanProperties: v1alpha1.InfinispanConnectionProperties{UseKogitoInfra: false}},
+			KafkaMeta:         v1alpha1.KafkaMeta{KafkaProperties: v1alpha1.KafkaConnectionProperties{UseKogitoInfra: true}},
```

Comment:
we need Kafka default for jobs-service ?

Reply:
To be able to communicate with Data Index, yes.

Reply:
I'll change to `false` then